### DISCUSSION TITLE
How to refresh the ssl certificates without restarting rabbitmq server and without dropping producer/consumer connections with tlsv1.3?

### DISCUSSION THREAD

**Discussion by AbyssWorld** [rabbitmq/rabbitmq-server#11997] [Questions] [answered]
How to refresh the ssl certificates without restarting rabbitmq server and without dropping producer/consumer connections with tlsv1.3?

**Comment by mkuratczyk** [comment]
Simply change the certificates (files). Erlang should pick up the new certificates and start using them for new connections.
It may take a moment (Erlang has a cache and doesn't check the files every single time it needs a cert). There's also a way to force Erlang to reload the certs by clearing the cache: https://www.rabbitmq.com/kubernetes/operator/using-operator#tls-conf (it's mentioned in the Kubernetes docs, but is applicable on other platforms; if it works for you, a PR to https://www.rabbitmq.com/docs/ssl to mention it would be appreciated)

**Reply by michaelklishin** [reply]
I will document this in the TLS guide since we now require an Erlang version that definitely has this behavior.

**Reply by michaelklishin** [reply]
https://github.com/rabbitmq/rabbitmq-website/pull/2016

**Reply by michaelklishin** [reply, marked as answer]
Should be available at https://www.rabbitmq.com/docs/ssl#rotation in a few minutes. If not, wait for a bit longer and clear your browser's cache.
